ServiceTier/Services/PresetService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Common.ModelsDTO;
+using Timer.DAL.Extensions;
+using Timer.DAL.Timer.DAL.Entities;
+using Timer.DAL.Timer.DAL.UnitOfWork;
+
+namespace ServiceTier.Services
+{
+    public class PresetService : BaseService, IPresetService
+    {
+        public PresetService(IUnitOfWork unitOfWork) : base(unitOfWork)
+        {
+
+        }
+
+        public void CreatePreset(PresetDTO presetDTO)
+        {
+            Preset preset = presetDTO.ToPreset();
+            unitOfWork.Presets.Create(preset);
+            unitOfWork.Save();
+            var presets = unitOfWork.Presets.GetAll();
+            foreach (var timer in presetDTO.Timers)
+            {
+                timer.PresetId = preset.Id;
+                unitOfWork.Timers.Create(timer.ToTimer());
+                unitOfWork.Save();
+            }
+        }
+
+        public void UpdatePreset(PresetDTO presetDTO)
+        {
+            Preset preset = presetDTO.ToPreset();
+            unitOfWork.Presets.Update(preset);
+            unitOfWork.Save();
+        }
+
+        public void UpdateTimer(TimerDTO timerDTO)
+        {
+            Timer.DAL.Timer.DAL.Entities.Timer timer = timerDTO.ToTimer();
+            unitOfWork.Timers.Update(timer);
+            unitOfWork.Save();
+        }
+
+        public void CreateTimer(TimerDTO timerDTO)
+        {
+            if(unitOfWork.Presets.GetByID(timerDTO.PresetId)==null)
+            {
+                throw new Exception("such preset doesn't exist");
+            }
+            unitOfWork.Timers.Create(timerDTO.ToTimer());
+            unitOfWork.Save();
+        }
+
+        public void DeletePresetById(int presetid)
+        {
+            if (unitOfWork.Presets.GetByID(presetid) == null)
+            {
+                throw new Exception("such preset doesn't exist");
+            }
+            if (unitOfWork.Presets.GetByID(presetid).UserId == null)
+            {
+                throw new Exception("this preset is standard");
+            }
+            unitOfWork.Presets.Delete(presetid);
+            unitOfWork.Save();
+        }
+
+        public PresetDTO GetPresetById(int presetid)
+        {
+            if (unitOfWork.Presets.GetByID(presetid) == null)
+            {
+                throw new Exception("such preset doesn't exist");
+            }
+            var preset = unitOfWork.Presets.GetByID(presetid);
+            var timers = unitOfWork.Timers.GetAll();
+            List<TimerDTO> timerDTOs = new List<TimerDTO>();
+            foreach (var t in timers)
```

Comment:
don't ever use one char variables. even in linq or foreach. it doesn't hurt to use whole name instead, and it drastically improves readability.